File: Demo/Objective-C/APSKeychainWrapperTests/APSKeychainWrapperTests/ViewController.m
//
//  ViewController.m
//  APSKeychainWrapperTests
//
//  Created by Hans Knoechel on 21/11/2016.
//  Copyright © 2016 Appcelerator. All rights reserved.
//

#import "ViewController.h"
#import "APSKeychainWrapper.h"

@interface ViewController () <APSKeychainWrapperDelegate>
@property(nonatomic,strong) APSKeychainWrapper* keychainItem;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _keychainItem = [[APSKeychainWrapper alloc] initWithIdentifier:@"password"
                                                               service:@"myservice"
                                                           accessGroup:@"$(AppIdentifierPrefix)com.appcelerator.APSKeychainWrapperTests"];
    
    [_keychainItem setDelegate:self];
}

- (IBAction)saveToKeychain {
    [_keychainItem save:@"my_secret_password"];
}

- (IBAction)readFromKeychain {
    [_keychainItem exists:^(BOOL exists, NSError *error) {
        if (error) {
            NSLog(@"Error reading from keychain: %@", [error localizedDescription]);
            return;
        }
        
        if (exists) {
            [_keychainItem read];
        }
    }];
}

- (IBAction)updateToKeychain {
    [_keychainItem update:@"my_new_password"];
}

- (IBAction)deleteFromKeychain {
    [_keychainItem reset];
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didReadValueWithError:(NSError *)error {
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didReadValueWithResult:(NSDictionary *)result {

}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didSaveValueWithError:(NSError *)error {
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didSaveValueWithResult:(NSDictionary *)result {
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didUpdateValueWithError:(NSError *)error {
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didUpdateValueWithResult:(NSDictionary *)result {
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didDeleteValueWithError:(NSError *)error
{
    
}

- (void)APSKeychainWrapper:(APSKeychainWrapper *)keychainWrapper didDeleteValueWithResult:(NSDictionary *)result {
    
}

@end

File: Demo/Swift/APSKeychainWrapperTests/APSKeychainWrapperTests/ViewController.swift
//
//  ViewController.swift
//  APSKeychainWrapperTests
//
//  Created by Hans Knoechel on 19/11/2016.
//  Copyright © 2016 Appcelerator. All rights reserved.
//

import UIKit

class ViewController: UIViewController, APSKeychainWrapperDelegate {
    
    var keychainItem : APSKeychainWrapper!

    override func viewDidLoad() {
        super.viewDidLoad()

        keychainItem = APSKeychainWrapper(identifier: "password", service: "myservice", accessGroup: "$(AppIdentifierPrefix)com.appcelerator.APSKeychainWrapperTests")
        keychainItem.delegate = self
    }
    
    @IBAction func saveToKeychain() {
        keychainItem.save("my_secret_password")
    }
    
    @IBAction func readFromKeychain() {
        keychainItem.exists({ (success: Bool, error: Error?) in
            if (error != nil) {
                print("Error reading from keychain: \(error?.localizedDescription)")
                return
            }
            
            if success {
                self.keychainItem.read()
            }
        })
    }
    
    @IBAction func updateToKeychain() {
        keychainItem.update("my_new_password")
    }
    
    @IBAction func deleteFromKeychain() {
        keychainItem.reset()
    }

    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didReadValueWithError error: Error!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didSaveValueWithError error: Error!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didDeleteValueWithError error: Error!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didUpdateValueWithError error: Error!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didReadValueWithResult result: [AnyHashable : Any]!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didSaveValueWithResult result: [AnyHashable : Any]!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didDeleteValueWithResult result: [AnyHashable : Any]!) {
        
    }
    
    func apsKeychainWrapper(_ keychainWrapper: APSKeychainWrapper!, didUpdateValueWithResult result: [AnyHashable : Any]!) {
        
    }
}

